router/src/test/java/io/micronaut/web/router/uri/UriUtilTest.java

```diff
@@ -0,0 +1,99 @@
+package io.micronaut.web.router.uri;
+
+import com.code_intelligence.jazzer.api.FuzzedDataProvider;
+import com.code_intelligence.jazzer.junit.FuzzTest;
+import org.junit.jupiter.api.Assertions;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+class UriUtilTest {
+    @FuzzTest(maxDuration = "30m")
```

Comment:
in addition to fuzz tests can we add some additional tests that run as part of the regular test suite? I couldn't find these.

Reply:
It's covered by existing tests (the malformed uri ones mostly), but I just added a check that request.uri works as expected